mipengine/node/udfgen/udfgenerator.py

```diff
@@ -175,15 +177,23 @@ def schema(self):
         return schema
 
     @classmethod
-    def from_table_info(cls, table_info):
-        try:
-            dtype = next(col.dtype for col in table_info.schema if col.name == "val")
-        except StopIteration:
-            raise TypeError("TableInfo doesn't have tensor-like schema.")
+    def from_table_info(cls, name, table_info):
+        dim_pattern = r"dim[0-9]+"
+        schema = table_info.schema
+        for col in schema:
+            if col.name == "val":
+                continue
+            if re.fullmatch(dim_pattern, col.name):
+                continue
+            if col.name == "row_id":
```

Comment:
Should "row_id" and "node_id" be global variables somewhere?

I am seeing these hard-coded in many places in the code.